src/idlcxx/tests/CMakeLists.txt

```diff
@@ -11,14 +11,50 @@
 #
 include(CUnit)
 
+set(out_dir_ ${CMAKE_CURRENT_BINARY_DIR})
+
+# Creates C resources file from files in given directory
+function(create_resources dir output)
+    # Create empty output file
+    file(WRITE ${out_dir_}/${output} "")
+    # Collect input files
+    file(GLOB bins ${dir}/*)
+    # Iterate through input files
+    foreach(bin ${bins})
+        # Get short filename
+        string(REGEX MATCH "([^/]+)$" filename ${bin})
+        # Replace filename spaces & extension separator for C compatibility
+        string(REGEX REPLACE "\\.| |-" "_" filename ${filename})
+        # Read hex data from file
+        file(READ ${bin} filedata HEX)
+        # Convert hex data for C compatibility
+        string(REGEX REPLACE "([0-9a-f][0-9a-f])" "0x\\1," filedata ${filedata})
+        #replace windows line endings (CR+LF) with linux line endings (LF)
+        string(REGEX REPLACE "0x0d,0x0a" "0x0a" filedata ${filedata})
+        # Add terminating NULL to have the array be interchangeable with a string
+        string(CONCAT filedata ${filedata} "0x0")
+        # Append data to output file
+        file(APPEND ${out_dir_}/${output} "const unsigned char ${filename}[] = {${filedata}};\nconst unsigned ${filename}_size = sizeof(${filename});\n")
+    endforeach()
+endfunction()
+
+create_resources(generator_test_cases generator_test_cases.h)
+
```

Comment:
Would be better to add this as a separate target so it's done at compile time not configure time. I think I'm also missing that the files are marked as configure files? Meaning that they target is recompiled if the contents of the files is changed.

Reply:
I will change the processed test files to configure files.